services/src/floodlight-service/floodlight-api/src/main/java/org/openkilda/floodlight/api/request/factory/FlowSegmentRequestProxiedFactory.java

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2019 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.api.request.factory;
+
+import org.openkilda.floodlight.api.request.FlowSegmentRequest;
+
+import java.util.UUID;
+
+public class FlowSegmentRequestProxiedFactory extends FlowSegmentRequestFactory<FlowSegmentRequest> {
```

Comment:
P.P.S. in the sample above my intent was to get rid of generics here, by replacing them with actual payload in successor classes, i.e.  messageContext, metadata, endpoint, encapsulation etc. if instead of building blank requests. If your original idea was not duplicating requests fields in factory classes, than generics of course should be OK. But anyway, it seems that introducing the proxy is an overkill. Could you share which warnings you are trying to solve by adding a proxy?